### ISSUE TITLE
Tips to get it working

### ISSUE THREAD

**Issue by verisgit** [free-dmo/free-dmo-stm32#30]
1) Buy a USB UART -> TTL cable at the start of the project and upload the firmware before you start all the other steps.
https://www.amazon.com.au/dp/B08761Z5K9
2) For firmware flashing you don't seem to be able to use the MicroUSB to upload it. You need to use the standard procedure and solder/jumper a USB UART interface or cable onto the BluePill. You don't need to connect all 4 pins to the board (you can but keep in mind one of them is used by later steps so you may have to solder). You only need to connect the TX, RX and ground wires because you can power it via the MicroUSB port - just plug that into a normal MicroUSB charger. But keep in mind the ground wire can just be jumpered onto the 4-pin ground wire sticking out the end of the blue pill temporarily whilst you are flashing the firmware. No real need to solder on a pin for the ground.
3) Check the colours on your JST GH 6pin cables are the same as the author. Mine were different hence nothing worked (cables purchased online seem to have different colour orders). I had to disconnect and swap the green and white wires, as mine were backwards compared to the ones the author used. You can use a needle and carefully disconnect the pins and change the order under a magnifying glass. You probably have one in your soldering station if you are soldering these tiny boards anyway. Once I did that, it all worked perfectly.

Troubleshooting:
-If you turn the printer on and insert paper, feed it and advance it to the first stop, and you get an error, then the main board is not connected to the JST GH 6pin cable (i.e. blue pill and RFID aren't connected).
-If you turn the printer on and it's not detecting any labels, you do the above, and it's still not detecting anything, then you have a dry solder somewhere on the blue pill (inspect them all with a magnifying glass), or one of the JST GH 6pin cables is not seated properly.
-When it works, you should see your printer detect your labels per whatever firmware options you selected (even if there's none in the printer). And when you print some, and then restart the printer, the count will reset to full. i.e. start at 120, print a label - now 119, restart it goes back to 120.

**Comment by Mathero11** [comment]
The bluepill board only gets 2.1V from the mainboard. (im using a 5XL printer)
When connected only from the bluepill to the mainboard it powers on, when connecting to the nfc scanner the red LED stops (its at 1.4V now)
After giving it some extra power by USB, it stays on.
Except that the dymo software doesn't read any data.

I have triple checked all the wires, they are correct, no dry solder, no touching of more pins.

You know why?

**Comment by verisgit** [comment]
I’d say the wires are wrong, ignore the colours and trace the pins. I had weird issues with mine and it was because the manufacturer I got my cable from used a different colour sequence so some of the wires got mixed up as I initially didn’t really check the pins just wired it up in sequence one wire was in the wrong spot. 

Mines a 5xl also and works fine.

**Comment by Mathero11** [comment]
You were correct. I printed the picture of the connector on the mainboard, but it flipped the image. So I had wrong pin layout.

Now with the stm board in place, the dmo software recognizes the label for about 4 seconds, after that it prints the message: Unkown label type.

**Comment by aptura-au** [comment]
When you flashed the STM did you flash it with a default label that actually exists in your region? The Dymo software sometimes doesn't recognise international ones and could be freaking out because it's not sure what's installed.

That said, it should be recognising it then resetting it again. Not sure about that one, maybe ask the author.

**Comment by Mathero11** [comment]
Yes you are right. The ones without S0 in the beginning (like 30332) doesn't work on this version.
I bought 28x89 S072370 and put that block inside the code, now it recognizes it.
